TC-SC-4.10 python test

raul-marquez-csa opened the pull request:
**Category:**
Functional conformance

**Description:**
The purpose of this test case is to verify that a Short Idle Time ICD node properly advertises SAI/SII
values.

**Verifies:**
  - SII key is higher than the SESSION_IDLE_INTERVAL default value (500 milliseconds)
  - SII key and SAI key is less than 1 hour (3600000 milliseconds)

**Full test plan link for details:**
https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/securechannel.adoc#tc-sc-4-10-operational-discovery-sit-icd-node-dut_commissionee

raul-marquez-csa committed:
Initial commit

raul-marquez-csa committed:
Gets TXT record properties

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Refactors getting txt record into class

raul-marquez-csa committed:
Merge branch 'tc-sc-4.10-python-test' of https://github.com/raul-marquez-csa/connectedhomeip into tc-sc-4.10-python-test

raul-marquez-csa committed:
Adds refresh option to TxtRecordUtil.get()

raul-marquez-csa committed:
TC complete, refactor TxtRecordUtil to MdnsHelper

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Fix restyle

raul-marquez-csa committed:
Fix restyle

raul-marquez-csa committed:
Fix lint, adds description to getOperationalServiceInfo

raul-marquez-csa committed:
Adds tc to tests.yaml, rewords verification texts

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Adds more description comments

raul-marquez-csa committed:
Merge branch 'tc-sc-4.10-python-test' of https://github.com/raul-marquez-csa/connectedhomeip into tc-sc-4.10-python-test

raul-marquez-csa committed:
Fix lint

raul-marquez-csa committed:
Restores tests.yaml formatting

raul-marquez-csa committed:
Updates tests.yaml to make TC_SC_4_10 run the lock app

raul-marquez-csa committed:
Adds zeroconf python dependency to Dockerfile

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

cecille commented on src/python_testing/mdns_helper.py:
```diff
@@ -0,0 +1,159 @@
+        pass
+
+
+class MdnsHelper:
```
Can you change the name of this to indicate this is for operational?

I think some of this logic will end up being common between operational and commissionable, so I think eventually you'll end up with a generic base class that exposes an operational and commissionable (and maybe commissioner) option. Anyway, for now, if you just change the name, you can extract the common bits later.

cecille commented on src/python_testing/mdns_helper.py:
```diff
@@ -0,0 +1,159 @@
+
+
+class MdnsHelper:
+    def __init__(self, dut):
```
what is dut here? Just the ID? Can you add a type indicator maybe a doc comment?

cecille commented on src/python_testing/mdns_helper.py:
```diff
@@ -0,0 +1,159 @@
+        value = self._service_info.properties[byte_key]
+
+        # Check if the value is not None
+        asserts.assert_is_not_none(value, f"Value for property '{key}' not present")
```
Can you please move this external to the class? That will make this class more generally useful for other tests in the future.

cecille commented on src/python_testing/mdns_helper.py:
```diff
@@ -0,0 +1,159 @@
+        property values based on the provided key.
+
+        Args:
+            dut: The device under test. It should contain attributes like `dut_node_id` and
```
dut isn't passed in as a parameter here.

cecille commented on src/python_testing/mdns_helper.py:
```diff
@@ -0,0 +1,159 @@
+                name=self._name,
+                type_=MDNS_TYPE_OPERATIONAL
+            )
+            asserts.assert_is_not_none(service_info, f"Service info for {self._name} not found")
```
same here.

raul-marquez-csa committed:
Adds zeroconf extra packages to tests.yaml, restores Dockerfile and version files

raul-marquez-csa committed:
Merge branch 'tc-sc-4.10-python-test' of https://github.com/raul-marquez-csa/connectedhomeip into tc-sc-4.10-python-test

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

andy31415 commented on src/python_testing/mdns_helper.py:
Please avoid names that contain filler words such as helper/support/util or similar.

Could we define what this file intends to do without such naming? This applies to both file name and class names. maybe something like `MatterMdnsRecordValidation` for class name (and translate this into a filename as well) or just mdns_validation or similar.

andy31415 commented on src/python_testing/mdns_helper.py:
```diff
@@ -0,0 +1,159 @@
+MDNS_TYPE_OPERATIONAL = "_matter._tcp.local."
+
+'''
+A service listener for the TXT record data to get populated
```
doc comments should follow class definition. How about `EmptyServiceListener` or `NoopServiceListener` ?

andy31415 commented on src/python_testing/mdns_helper.py:
```diff
@@ -0,0 +1,159 @@
+        # Convert the value from bytes to string
+        return value.decode('utf-8')
+
+    async def getTxtRecord(self, key: str = None, refresh=False):
```
could we type the return value as well?

andy31415 replied:
Type of key should be `Optional[str]` if the value can be none

andy31415 commented on src/python_testing/TC_SC_4_10.py:
```diff
@@ -0,0 +1,72 @@
+
+        # Get SAI/SII values
+        mh = MdnsHelper(self)
+        SAI_MS = int(await mh.getTxtRecord(key="SAI"))
```
I would lowercase these as `sai_ms` and `sii_ms` since these are not constants (as opposed to the others)

raul-marquez-csa committed:
Implements latest review comments

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Merge branch 'tc-sc-4.10-python-test' of https://github.com/raul-marquez-csa/connectedhomeip into tc-sc-4.10-python-test

raul-marquez-csa committed:
Fix restyle

raul-marquez-csa committed:
Updates _get_txt_record_key_value

raul-marquez-csa committed:
Fix lint/restyle

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Merge branch 'tc-sc-4.10-python-test' of https://github.com/raul-marquez-csa/connectedhomeip into tc-sc-4.10-python-test

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Adds chip_enable_icd_server=true for lock app gn command
Renames GnBuildArgs to GnBuildArgsLocal in gn.py as it was being overriden by HostBuilder in host.py

raul-marquez-csa committed:
Merge branch 'tc-sc-4.10-python-test' of https://github.com/raul-marquez-csa/connectedhomeip into tc-sc-4.10-python-test

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Move logic to host.py

raul-marquez-csa committed:
Merge branch 'tc-sc-4.10-python-test' of https://github.com/raul-marquez-csa/connectedhomeip into tc-sc-4.10-python-test

raul-marquez-csa committed:
Restore logic

raul-marquez-csa committed:
Restore

raul-marquez-csa committed:
Fix context

andy31415 committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Re-locate logic

raul-marquez-csa committed:
Merge branch 'tc-sc-4.10-python-test' of https://github.com/raul-marquez-csa/connectedhomeip into tc-sc-4.10-python-test

raul-marquez-csa committed:
Adds logic

raul-marquez-csa committed:
Fix restyle

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa committed:
Merge branch 'master' into tc-sc-4.10-python-test

raul-marquez-csa commented:
This test case has been deprecated and has been included in https://github.com/project-chip/connectedhomeip/pull/31982
